File: app/src/main/java/ardjomand/leonardo/nutrimeal/cart/CartContract.java
package ardjomand.leonardo.nutrimeal.cart;

import ardjomand.leonardo.nutrimeal.data.pojos.CartMeal;
import ardjomand.leonardo.nutrimeal.data.pojos.Meal;

interface CartContract {

    interface View {

        void addCartMeal(CartMeal meal);

        void updateCartMeal(CartMeal meal);

        void showEmptyMeals();

        void showError();
    }

    interface Presenter {

        void setView(View view);

        void subscribeToCartUpdates();

        void unsubscribeFromCartUpdates();

        void addMealToCart(Meal meal);

        void placeOrder();
    }
}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/cart/CartContract.kt
package ardjomand.leonardo.nutrimeal.cart

import ardjomand.leonardo.nutrimeal.data.pojos.CartMeal
import ardjomand.leonardo.nutrimeal.data.pojos.Meal

internal interface CartContract {

    interface View {

        fun addCartMeal(cartMeal: CartMeal)

        fun updateCartMeal(cartMeal: CartMeal)

        fun showEmptyMeals()

        fun showError()
    }

    interface Presenter {

        fun setView(view: View?)

        fun subscribeToCartUpdates()

        fun unsubscribeFromCartUpdates()

        fun addMealToCart(meal: Meal)

        fun placeOrder()
    }
}
